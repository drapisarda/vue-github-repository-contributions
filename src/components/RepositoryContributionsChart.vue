<template>
  <highcharts :options="chartOptions" />
</template>

<script>
export default {
  name: 'RepositoryContributionsChart',
  props: {
    contributors: {
      type: Array,
      required: false,
      default: () => ([]),
    },
    title: {
      type: String,
      required: false,
      default: '',
    },
  },
  computed: {
    chartOptions() {
      const names = this.contributors.map(_ => _.name);
      const counts = this.contributors.map(_ => _.count);

      return {
        title: {
          text: this.title,
        },
        yAxis: {
          title: {
            text: 'Contributions',
          },
        },
        xAxis: {
          categories: names,
          title: {
            text: 'Contributors',
          },
        },
        series: [{
          type: 'column',
          data: counts,
          showInLegend: false,
          tooltip: {
            pointFormat: '',
            headerFormat: '<span>{point.key} - <b>{point.y}</b></span><br/>',
          },
        }],
      };
    },
  },
};
</script>
