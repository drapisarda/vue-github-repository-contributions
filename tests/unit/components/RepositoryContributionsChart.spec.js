import { shallowMount, createLocalVue } from '@vue/test-utils';
import HighchartsVue from 'highcharts-vue';
import RepositoryContributionsChart from '@/components/RepositoryContributionsChart.vue';

const localVue = createLocalVue();
localVue.use(HighchartsVue);

describe('RepositoryContributionsChart', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(RepositoryContributionsChart, {
      localVue,
    });
  });

  it('render component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('maps title prop to chart title', () => {
    const title = 'angular/angular';

    wrapper.setProps({
      title,
    });

    expect(wrapper.vm.chartOptions.title.text).toBe(title);
  });

  it('maps contributors prop to chart axis', () => {
    const contributors = [
      {
        name: 'name1',
        count: 10,
      },
      {
        name: 'name2',
        count: 15,
      },
    ];
    const names = contributors.map(_ => _.name);
    const counts = contributors.map(_ => _.count);

    wrapper.setProps({
      contributors,
    });

    expect(wrapper.vm.chartOptions.xAxis.categories).toEqual(names);
    expect(wrapper.vm.chartOptions.series[0].data).toEqual(counts);
  });
});
