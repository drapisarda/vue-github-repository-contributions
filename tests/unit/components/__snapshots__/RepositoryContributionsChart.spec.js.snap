// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoryContributionsChart render component 1`] = `<highcharts-stub constructortype="chart" options="[object Object]" updateargs="true,true"></highcharts-stub>`;
